integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioGateway.java

```diff
@@ -0,0 +1,460 @@
+    logger.info("Istio Ingress Port is {0}", istioIngressPort);
+
+    // We can not verify Rest Management console thru Adminstration NodePort
+    // in istio, as we can not enable Adminstration NodePort
+    if (!WEBLOGIC_SLIM) {
+      String consoleUrl = "http://" + K8S_NODEPORT_HOST + ":" + istioIngressPort + "/console/login/LoginForm.jsp";
+      boolean checkConsole =
+          checkAppUsingHostHeader(consoleUrl, domainNamespace + ".org");
+      assertTrue(checkConsole, "Failed to access WebLogic console");
+      logger.info("WebLogic console is accessible");
+    } else {
+      logger.info("Skipping WebLogic console in WebLogic slim image");
+    }
+
+    Path archivePath = Paths.get(ITTESTS_DIR, "../operator/integration-tests/apps/testwebapp.war");
+    ExecResult result = null;
+    result = deployToClusterUsingRest(K8S_NODEPORT_HOST,
+        String.valueOf(istioIngressPort),
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT,
+        clusterName, archivePath, domainNamespace + ".org", "testwebapp");
+    assertNotNull(result, "Application deployment failed");
+    logger.info("Application deployment returned {0}", result.toString());
+    assertEquals("202", result.stdout(), "Deployment didn't return HTTP status code 202");
+
+    String url = "http://" + K8S_NODEPORT_HOST + ":" + istioIngressPort + "/testwebapp/index.jsp";
+    logger.info("Application Access URL {0}", url);
+
+    return istioIngressPort;
+  }
+
+  private Map<String, String> processHttpRequest(String curlUrlPath,
+                                                 String headerOption) {
+    String[] httpAttrArray = {"sessioncreatetime", "sessionid", "primary", "secondary", "count"};
+    Map<String, String> httpAttrInfo = new HashMap<String, String>();
+
+    // build curl command
+    String curlCmd = buildCurlCommand(curlUrlPath, headerOption);
+    logger.info("==== Command to set HTTP request and get HTTP response {0} ", curlCmd);
+
+    // set HTTP request and get HTTP response
+    ExecResult execResult = assertDoesNotThrow(
+        () -> execCommand(domainNamespace, adminServerPodName,
+        null, true, "/bin/sh", "-c", curlCmd));
+    if (execResult.exitValue() == 0) {
+      logger.info("\n ===== HTTP response is \n " + execResult.stdout());
+      assertAll("Check that primary server name is not null or empty",
+          () -> assertNotNull(execResult.stdout(), "Primary server name shouldn’t be null"),
+          () -> assertFalse(execResult.stdout().isEmpty(), "Primary server name shouldn’t be  empty")
+      );
+
+      for (String httpAttrKey : httpAttrArray) {
+        String httpAttrValue = getHttpResponseAttribute(execResult.stdout(), httpAttrKey);
+        httpAttrInfo.put(httpAttrKey, httpAttrValue);
+      }
+    } else {
+      fail("==== Failed to process HTTP request " + execResult.stderr());
+    }
+
+    return httpAttrInfo;
+  }
+
+  private String buildCurlCommand(String curlUrlPath, String headerOption) {
```

Comment:
buildCurlCommand method is repeated in both class. Can be extracted out to a SessionMigration utility class to be used in both istio and non-istio usecases.

Reply:
changed